ydb/_topic_reader/topic_reader_sync.py

```diff
@@ -86,6 +86,57 @@ def async_wait_message(self) -> concurrent.futures.Future:
 
         return self._caller.unsafe_call_with_future(self._async_reader.wait_message())
 
+    def _make_batch_slice(
+        self,
+        batch: Union[PublicBatch, None],
+        max_messages: typing.Union[int, None] = None,
+        max_bytes: typing.Union[int, None] = None,
+    ) -> Union[PublicBatch, None]:
+        all_amount = float("inf")
+
+        # A non-empty batch must stay non-empty regardless of the max messages value
+        if max_messages is not None:
+            max_messages = max(max_messages, 1)
+        else:
+            max_messages = all_amount
+
+        if max_bytes is not None:
+            max_bytes = max(max_bytes, 1)
+        else:
+            max_bytes = all_amount
+
+        is_batch_set = batch is not None
+        is_msg_limit_set = max_messages < all_amount
```

Comment:
why do you need all_amount instead check `max_messages is not None`?

Reply:
Because `max_messages` is being set to `all_amount` ([up above](https://github.com/ydb-platform/ydb-python-sdk/pull/431/files#diff-4de4f866b7b2f3095a8ebb8ed96e505820fc491d363a5504162343ddad249c5fR101)) in case if it hasn't been provided (e.g. it's `None`).